### PR TITLE
[MER-2481] My courses view for instructors

### PR DESCRIPTION
**Pull request by nicocirio**
[Link](https://eliterate.atlassian.net/browse/MER-2481) to the ticket

https://github.com/Simon-Initiative/oli-torus/assets/74839302/460de448-c09e-4f04-bca1-af0efcf60a1b

This PR also implements `verified_routes` allowing us to use the new sigil `~p` to define URLs. 
More info about this feature can be found [here](https://phoenixframework.org/blog/phoenix-1.7-final-released)

### CODE REVIEW

File: lib/oli_web.ex [hunk 1]
```diff
@@ -137,6 +139,18 @@ defmodule OliWeb do
...
+      unquote(verified_routes())
+    end
+  end
+
+  def verified_routes do
+    quote do
+      use Phoenix.VerifiedRoutes,
+        endpoint: OliWeb.Endpoint,
+        router: OliWeb.Router,
+        statics: OliWeb.static_paths()
```
Comment (augusto1024) [inline lib/oli_web.ex:153]: ❤️

File: lib/oli_web/components/delivery/instructor_dashboard.ex [hunk 2]
```diff
@@ -74,21 +82,23 @@ defmodule OliWeb.Components.Delivery.InstructorDashboard do
...
-        </ul>
-      </div>
+                "}
+            >
+              <%= if is_function(label), do: label.(), else: label %>
+              <%= if badge do %>
+                <span class="text-xs inline-block py-1 px-2 ml-2 leading-none text-center whitespace-nowrap align-baseline font-bold bg-delivery-primary text-white rounded">
+                  <%= badge %>
+                </span>
+              <% end %>
```
Comment (augusto1024) [inline lib/oli_web/components/delivery/instructor_dashboard.ex:92]: NIT: This looks like a good opportunity to replace that `<%= if badge do %>` for LiveView's new `:if` decorator 😁

File: lib/oli_web/components/delivery/instructor_dashboard.ex [hunk 3]
```diff
@@ -53,14 +58,17 @@ defmodule OliWeb.Components.Delivery.InstructorDashboard do
...
-            <li class="nav-item" role="presentation">
-              <.link patch={path}
-                class={"
+    <div class="container mx-auto my-4">
+      <ul
+        class="nav nav-tabs flex flex-col md:flex-row flex-wrap list-none border-b-0 pl-0 mb-4"
+        id="tabs-tab"
+        role="tablist"
+      >
+        <%= for %TabLink{label: label, path: path, badge: badge, active: active} <- @tabs do %>
```
Comment (augusto1024) [inline lib/oli_web/components/delivery/instructor_dashboard.ex:67]: NIT: This looks like a good opportunity to replace that `<%= for... do %>` for LiveView's new `:for` decorator 😁

File: lib/oli_web/controllers/pow/registration_html/edit.html.heex [hunk 4]
```diff
@@ -102,7 +102,7 @@
         <%= hidden_input f, :link_account, value: @link_account %>
       <% end %>
       <%= submit "Update", class: "btn btn-md btn-primary btn-block" %>
-      <%= link "Cancel", to: value_or(assigns[:cancel_path], Routes.delivery_path(@conn, :open_and_free_index)), class: "btn btn-md btn-outline-secondary btn-block mt-3" %>
+      <%= link "Cancel", to: value_or(assigns[:cancel_path], Routes.live_path(@conn, OliWeb.Delivery.OpenAndFreeIndex)), class: "btn btn-md btn-outline-secondary btn-block mt-3" %>
```
Comment (augusto1024) [inline lib/oli_web/controllers/pow/registration_html/edit.html.heex:105]: Can't we change this for `~p"/sections"` like in the other files?

File: lib/oli_web/live/delivery/open_and_free_index.ex [hunk 5]
```diff
@@ -0,0 +1,179 @@
+defmodule OliWeb.Delivery.OpenAndFreeIndex do
+  use OliWeb, :live_view
+
+  on_mount({OliWeb.LiveSessionPlugs.SetCurrentUser, :with_preloads})
+  on_mount(OliWeb.LiveSessionPlugs.SetCtx)
```
Comment (augusto1024) [inline lib/oli_web/live/delivery/open_and_free_index.ex:5]: ❤️

File: lib/oli_web/live/delivery/open_and_free_index.ex [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+          s.id in ^section_ids and s.status == :active and e.status == :enrolled and
+            ecr.context_role_id == ^instructor_context_role_id,
+        preload: [:user],
+        select: {s.id, e}
+      )
+
+    Repo.all(query)
+    |> Enum.group_by(fn {section_id, _} -> section_id end, fn {_, enrollment} ->
+      Utils.user_name(enrollment.user)
+    end)
```
Comment (augusto1024) [inline lib/oli_web/live/delivery/open_and_free_index.ex:157]: WDYT about moving this query to the `lib/oli/delivery/sections.ex` context file?

**Review by augusto1024 — COMMENTED** [review]
Great job! I really love the `verified_routes` thing (i hate having to go to the router and see how I can access the route using the routes helper 😠).

I left some minor comments 😁

**Review by augusto1024 — APPROVED** [review]
Thanks for the changes 🙌

File: lib/oli/delivery/sections.ex [hunk 7]
```diff
@@ -1020,6 +1025,39 @@ defmodule Oli.Delivery.Sections do
...
+
+    query =
+      from(
+        e in Enrollment,
+        join: s in Section,
+        on: e.section_id == s.id,
+        join: ecr in EnrollmentContextRole,
+        on: e.id == ecr.enrollment_id,
+        where:
+          s.id in ^section_ids and s.status == :active and e.status == :enrolled and
```
Comment (darrensiegel) [inline lib/oli/delivery/sections.ex:1049]: We shouldn't be including `s.status == :active` here.  If client code is passing in a list of section_ids, that client code might not be making the same assumptions that they only want results to return if the section is "active".

**Review by darrensiegel — CHANGES_REQUESTED** [review]

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**